### PR TITLE
New uitype MultiEmail

### PR DESCRIPTION
**Pull request by arek1-618**
New uitype MultiEmail
chrome_2018-08-20_14-00-26

### CODE REVIEW

**Comment by arek1-618** [comment]
chrome_2018-08-23_13-36-34

File: app/QueryField/MultiEmailField.php [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<?php
+/**
+ * MultiEmail Query Field Class.
+ *
+ * @copyright YetiForce Sp. z o.o
+ * @license   YetiForce Public License 3.0 (licenses/LicenseEN.txt or yetiforce.com)
+ * @author    Arkadiusz Adach <a.adach@yetiforce.com>
+ */
+namespace \App\QueryField;
+
```
Comment (mariuszkrzaczkowski) [inline app/QueryField/MultiEmailField.php]: ?

File: layouts/basic/modules/Vtiger/Edit/Field/MultiEmail.tpl [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+	<div class="tpl-Edit-Field-MultiEmail d-flex align-items-center js-multi-email">
+		<input class="js-hidden-email" data-js="value" type="hidden" name="{$FIELD_MODEL->getFieldName()}" value=""/>
+		<button class="btn btn-outline-success border mr-2 mb-2 h-100 js-add-item"
+				data-js="click" type="button">
+			<span class="fas fa-plus" title="{\App\Language::translate('LBL_ADD', $MODULE)}"></span>
+		</button>
+		<div class="form-inline">
+			{counter start=0 skip=1 print=false}
+			{foreach item=ITEM from=$NOT_DISPLAY_LIST_VALUES}
+				{include file=\App\Layout::getTemplatePath('Edit/Field/MultiEmailValue.tpl', 'Vtiger')}
```
Comment (mariuszkrzaczkowski) [inline layouts/basic/modules/Vtiger/Edit/Field/MultiEmail.tpl]: {include file=\App\Layout::getTemplatePath('Edit/Field/MultiEmailValue.tpl', $MODULE)}

File: layouts/basic/modules/Vtiger/Edit/Field/MultiEmailValue.tpl [hunk 3]
```diff
@@ -0,0 +1,47 @@
+{*<!-- {[The file is published on the basis of YetiForce Public License 3.0 that can be found in the following directory: licenses/LicenseEN.txt or yetiforce.com]} -->*}
+
```
Comment (mariuszkrzaczkowski) [inline layouts/basic/modules/Vtiger/Edit/Field/MultiEmailValue.tpl]: ?

File: layouts/basic/modules/Vtiger/Edit/Field/MultiEmail.tpl [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+{strip}
+	{assign var="FIELD_INFO" value=\App\Purifier::encodeHtml(\App\Json::encode($FIELD_MODEL->getFieldInfo()))}
+	{assign var="SPECIAL_VALIDATOR" value=$FIELD_MODEL->getValidator()}
+	{assign var=FIELD_VALUE value=$FIELD_MODEL->getEditViewDisplayValue($FIELD_MODEL->get('fieldvalue'),$RECORD)}
+	{if !empty($FIELD_MODEL->get('fieldvalue'))}
+		{assign var=NOT_DISPLAY_LIST_VALUES value=$FIELD_VALUE}
+	{else}
+		{assign var=NOT_DISPLAY_LIST_VALUES value=[['e'=>'']]}
+	{/if}
+	<div class="tpl-Edit-Field-MultiEmail d-flex align-items-center js-multi-email">
```
Comment (mariuszkrzaczkowski) [inline layouts/basic/modules/Vtiger/Edit/Field/MultiEmail.tpl]: tpl-Base-Edit-Field-MultiEmail

File: app/QueryField/MultiEmailField.php [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+/**
+ * MultiEmail Query Field Class.
+ *
+ * @copyright YetiForce Sp. z o.o
+ * @license   YetiForce Public License 3.0 (licenses/LicenseEN.txt or yetiforce.com)
+ * @author    Arkadiusz Adach <a.adach@yetiforce.com>
+ */
+
+namespace App\QueryField;
+
```
Comment (mariuszkrzaczkowski) [inline app/QueryField/MultiEmailField.php]: ?

File: layouts/basic/modules/Vtiger/Edit/Field/MultiEmailValue.tpl [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+		<label for="staticEmail2" class="sr-only" for="email-value">
+			{\App\Language::translate('LBL_EMAIL_ADRESS', $MODULE)}
+		</label>
+		<div class="input-group">
+			<div class="input-group-prepend">
+				<button class="btn btn-outline-danger border js-remove-item" data-js="click" type="button">
+					<span class="fas fa-times" title="{\App\Language::translate('LBL_REMOVE', $MODULE)}"></span>
+				</button>
+			</div>
+			<input type="text" class="form-control js-email" data-js="email" id="email-value"
```
Comment (mariuszkrzaczkowski) [inline layouts/basic/modules/Vtiger/Edit/Field/MultiEmailValue.tpl]: YF33

**Pull request merged**